numpy/core/src/multiarray/datetime_strings.c

```diff
@@ -323,22 +326,37 @@ parse_iso_8601_datetime(char const *str, Py_ssize_t len,
     if (len == 3 && tolower(str[0]) == 'n' &&
                     tolower(str[1]) == 'o' &&
                     tolower(str[2]) == 'w') {
-        NPY_TIME_T rawtime = 0;
         PyArray_DatetimeMetaData meta;
 
-        time(&rawtime);
+        // initialize the _PyTime_t struct
+        _PyTime_t rawtime = _PyTime_GetSystemClock();
```

Comment:
I was getting segfaults when I set this just to some arbitrary positive `int64_t` value like `0` or `1` I think. Maybe setting it to some reasonably large integer offset from the epoch might be safe and avoid two time draws in the control flow?